selfservice/strategy/oidc/provider_microsoft.go

```diff
@@ -53,25 +50,13 @@ func (m *ProviderMicrosoft) OAuth2(ctx context.Context) (*oauth2.Config, error)
 	return m.oauth2ConfigFromEndpoint(ctx, endpoint), nil
 }
 
-func (m *ProviderMicrosoft) Claims(ctx context.Context, exchange *oauth2.Token, query url.Values) (*Claims, error) {
+func (m *ProviderMicrosoft) Claims(ctx context.Context, exchange *oauth2.Token, _ url.Values) (*Claims, error) {
 	raw, ok := exchange.Extra("id_token").(string)
 	if !ok || len(raw) == 0 {
 		return nil, errors.WithStack(ErrIDTokenMissing)
 	}
 
-	parser := new(jwt.Parser)
```

Comment:
From what I see it was only used to construct the issuer URL, but we actually have all the relevant information already in the config.
Now that we moved the cache, it is not necessary anymore, but I feel like it still does make sense to avoid this complex and potentially insecure codepath. Happy to move to a separate PR.

Reply:
Yeah, I see it does make sense to remove this. Can you please check that this doesn't add regressions? I don't think we have any tests for this

Reply:
OK, I'll revert this now as it seems to break the common tenant. Will investigate separately.